### PR TITLE
feat(overview): calibrate synthesis prompt for evidence-grounded headlines

### PR DESCRIPTION
**Pull request by lvndry**
## What this PR delivers

Tightens `PRODUCT_OVERVIEW_PROMPT` so the synthesis LLM produces headlines and summaries that match document evidence — aligned with the same rules the LLM quality review guard uses (`UNSUPPORTED_CLAIMS`, signal strength, etc.).

Complements #192 (validation retry loop) by reducing how often overclaims happen on the first attempt.

## Why this matters

**For operators:** Fewer overview guard rejections and retries during tier regen — Netflix-style failures from overstrong headlines should drop.

**For maintainers:** Prompt examples no longer train the model toward "sells" + "indefinitely" wording that guards reject when signals/citations are weaker.

## How overview synthesis behaviour changes

| Path | Change |
|---|---|
| First-pass overview generation | System prompt adds calibration rules: decide `privacy_signals` first, then constrain headline/summary strength. |
| Products with ad sharing but no explicit sale | Model instructed to use "shares with advertisers" not "sells". |
| Thin or ambiguous evidence | Model instructed to prefer `headline_claim: null` over worst-case inference. |
| Validation retry loop (#192) | Unchanged — still catches remaining overclaims. |

## UX changes operators will notice

- Product pages may show slightly more measured headlines (e.g. "shares with ad partners" vs "sells your data") when documents support the weaker claim only.
- Pipeline logs should show fewer `UNSUPPORTED_CLAIMS` validation failures on first attempt.

## Tests

- Pre-commit ruff + ty + backend tests pass (prompt-only change).
- No new automated tests — prompt behaviour validated via regen spot-checks.

### Edge cases to verify in a staging session

1. Regenerate Netflix — expected: headline passes guard on attempt 1 or 2 without claiming voice recordings/indefinite retention unless evidenced.
2. Regenerate a product with explicit data sale language — expected: headline may still use "sells" when `sells_data` is "yes".
3. Product with only one core doc — expected: null headline or cautious summary when evidence is thin.

## Notes for reviewers

- Good/bad examples updated to remove the old Spotify "sells… indefinitely" training example.
- New **HEADLINE AND SUMMARY CALIBRATION** section mirrors `llm_review.py` UNSUPPORTED_CLAIMS criteria.
- Hard rule #7 added: calibrate prose strength to signals.

### CODE REVIEW

File: packages/backend/src/prompts/analysis_prompts.py [hunk 1]
```diff
@@ -304,20 +305,34 @@
...
 
+## HEADLINE AND SUMMARY CALIBRATION (match quality review)
+Before writing `headline_claim` or `summary`, decide `privacy_signals` from extractions and critical clauses. Then apply:
+- **Sells vs shares:** Use "sells" or "sold to brokers" only when documents say sale or valuable consideration, or `sells_data` is "yes". Otherwise use "shares with advertisers/partners" or "uses for targeted ads".
+- **Indefinite retention:** Use "indefinitely" or "forever" only when no retention period is stated or documents explicitly say unlimited retention. If a period or category-specific retention is stated, describe it accurately.
+- **Biometrics / voice / sensitive data:** Mention in headline/summary only when extractions or critical clauses show it is collected or used — not because it appears once as an optional feature or hypothetical.
+- **AI training:** Mention only when `ai_training_on_user_data` is "yes" or documents clearly describe training on user content.
+- **Topic signals:** If deterministic topic signals show "not_disclosed", "missing", or "unclear" for a topic, do not state that topic as a confirmed practice in the headline.
```
[the remark this answers is not in the document]
Comment (lvndry) [reply packages/backend/src/prompts/analysis_prompts.py]: Addressed in 4775719: documented the `Deterministic per-topic signals` array in the INPUT section (matches the user-message field from `generate_product_overview`) and replaced the calibration bullet with explicit `status` values (`missing`, `not_disclosed`, `ambiguous`) plus `privacy_signals` of `unclear`/`not_specified`.
